### PR TITLE
langinfo bindings and defs

### PR DESCRIPTION
**Pull request by mrhota**

### CODE REVIEW

**Comment by mrhota** [comment]
not done yet; I wanted to make sure basic types were correct

**Comment by mrhota** [comment]
I guess I could test this locally instead of wasting CI cycles :wink:

I just figured out how to run the build locally. Silly me!

File: libc-test/build.rs [hunk 1]
```diff
@@ -66,6 +66,7 @@ fn main() {
         cfg.header("dirent.h");
         if openbsd {
             cfg.header("sys/socket.h");
+        } else if !netbsd {
```
Comment (mrhota) [inline libc-test/build.rs]: I forgot to remove this.

**Comment by alexcrichton** [comment]
Thanks! Can you also squash the commits down? Other than that r=me

**Comment by mrhota** [comment]
ok, @alexcrichton, thanks for your patience. I think we're good to go!

**Comment by mrhota** [comment]
@alexcrichton r?

**Pull request merged**